skimage/morphology/_skeletonize_3d.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import division, print_function, absolute_import
+
+import numpy as np
+
+from ._skeletonize_3d_cy import _compute_thin_image
+
+
+def _prepare_image(img_in):
+    """Convert to a binary image, pad the it w/ zeros, and ensure it's 3D.
+    """
+    if img_in.ndim < 2 or img_in.ndim > 3:
+        raise ValueError('expect 2D, got ndim = %s' % img_in.ndim)
+
+    img = img_in.copy()
+    if img.ndim == 2:
+        img = img.reshape((1,) + img.shape)
+
+    intensity = img.max()
+
+    # normalize to binary
+    img[img != 0] = 1
```

Comment:
2D `morphology.skeletonize` function explicitly raises for anything other than 0 or 1:
```
    if not np.all(np.in1d(image.flat, (0, 1))):
        raise ValueError('Image contains values other than 0 and 1')
```